### ISSUE TITLE
Support of parameter objects

### ISSUE THREAD

**Issue by kobezzza** [google/closure-compiler#392]
When describing the properties of the object in the function is very convenient to use the syntax with multiple use of the @param tag, but I noticed that GCC though not throws warnings in this syntax, but does not support it fully

``` js
/**
 * @param {{r: number}} params
 */
function foo(params) {

}

foo({r: 'd'}); // warning, all fine


/**
 * @param {!Object} params
 * @param {number} params.r
 */
function foo(params) {

}

foo({r: 'd'}); // fail, must be warning too
```
